velox/exec/WindowPartition.h

```diff
@@ -107,9 +107,50 @@ class WindowPartition {
       vector_size_t* rawPeerStarts,
       vector_size_t* rawPeerEnds) const;
 
+  /// Sets in 'rawFrameBounds' the frame boundary for the k range
+  /// preceding/following frame.
+  /// 'isStartBound' : start or end boundary of the frame.
+  /// 'isPreceding'  : preceding or following boundary.
+  /// 'frameColumn' : column which has the range boundary for that row.
+  /// 'startRow' : starting row in the partition for this buffer computation.
+  /// 'numRows' : number of rows to compute buffer for.
+  /// 'rawPeerStarts' : buffer of peer row values for each row. If the range
+  /// column is null, then its peer row value is the frame boundary.
+  void computeKRangeFrameBounds(
+      bool isStartBound,
+      bool isPreceding,
+      column_index_t frameColumn,
+      vector_size_t startRow,
+      vector_size_t numRows,
+      const vector_size_t* rawPeerStarts,
+      vector_size_t* rawFrameBounds) const;
+
  private:
   bool compareRowsWithSortKeys(const char* lhs, const char* rhs) const;
 
+  // Searches for 'currentRow[frameColumn]' in 'orderByColumn' of rows between
+  // 'start' and 'end' in the partition. 'firstMatch' specifies if first or last
+  // row is matched.
+  template <bool isAscending>
+  vector_size_t searchFrameValue(
+      bool firstMatch,
+      vector_size_t start,
+      vector_size_t end,
+      vector_size_t currentRow,
+      column_index_t orderByColumn,
+      column_index_t frameColumn) const;
+
+  // Iterates over 'numBlockRows' and searches frame value for each row.
```

Comment:
naming: numBlockRows sounds strange

Reply:
I didn't want to use numRows as that was conflicting with numRows() function in WindowPartition. But have changed the variable naming now.